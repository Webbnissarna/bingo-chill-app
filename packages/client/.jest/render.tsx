import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
