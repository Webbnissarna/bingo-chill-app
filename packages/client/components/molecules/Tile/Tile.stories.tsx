import type { Story, StoryDefault } from "@ladle/react";
import Tile from ".";
import { SPOOKY_GHOST_IMAGE_BASE64 } from "@/.ladle/constants";

interface StoryProps {
  text: string;
  colors: string[];
}

function StoryComponent({ text, colors }: StoryProps) {
  return (
    <Tile
      text={text}
      icon={SPOOKY_GHOST_IMAGE_BASE64}
      highlightColors={colors ?? []}
      onClick={() => undefined}
    />
  );
}

export const Default: Story<StoryProps> = StoryComponent;

Default.args = {
  text: "Hello World",
  colors: [],
};

Default.argTypes = {
  text: {
    control: { type: "text" },
  },
  colors: {
    options: ["#bf616a", "#d08770", "#ebcb8b", "#a3be8c", "#b48ead"],
    control: { type: "check" },
  },
};

export default {
  title: "Molecules / Tile",
} satisfies StoryDefault;
