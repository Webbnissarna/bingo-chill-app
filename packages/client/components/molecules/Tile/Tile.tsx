import Base64Icon from "@/components/atoms/Base64Icon";
import { generateGradientCSS } from "@/utils/css";
import { useIf } from "@webbnissarna/bingo-chill-common/src/utils/functional";

interface TileProps {
  text: string;
  icon: string;
  highlightColors: string[];
  onClick: () => void;
}

export default function Tile({
  text,
  icon,
  highlightColors,
  onClick,
}: TileProps): JSX.Element {
  const hasColors = highlightColors.length > 0;
  return (
    <div
      style={useIf(hasColors, {
        background: generateGradientCSS(highlightColors),
      })}
      className="max-w-[120px] max-h-[120px] aspect-square rounded-2xl bg-polarNight-3 flex flex-col items-center py-1 gap-1 cursor-pointer hover:bg-polarNight-2 active:bg-polarNight-1 hover:opacity-90 active:opacity-80 transition select-none"
      onClick={onClick}
    >
      <div className="grow h-2/3">
        <div className="text-center w-full h-full flex flex-col justify-center">
          <span
            className={`${useIf(
              hasColors,
              "bg-polarNight-0/75",
            )} font-text text-snowStorm-2 text-xs sm:text-base leading-snug inline-block w-full max-h-full rounded px-2 py-1`}
          >
            {text}
          </span>
        </div>
      </div>
      <div className="w-1/3 h-1/3">
        <Base64Icon src={icon} description={text} />
      </div>
    </div>
  );
}
