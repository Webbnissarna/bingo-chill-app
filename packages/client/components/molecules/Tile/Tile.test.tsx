import { render } from "../../../.jest/render";
import Tile from ".";

describe("Tile", () => {
  it("renders", () => {
    const { asFragment } = render(
      <Tile text="" icon="" highlightColors={[]} onClick={() => undefined} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
