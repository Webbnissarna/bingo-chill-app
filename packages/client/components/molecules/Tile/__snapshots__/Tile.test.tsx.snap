// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders 1`] = `
<DocumentFragment>
  <div
    class="max-w-[150px] max-h-[150px] aspect-square rounded-2xl bg-polarNight-3 flex flex-col items-center py-2 gap-1 cursor-pointer hover:bg-polarNight-2 active:bg-polarNight-1 hover:opacity-90 active:opacity-80 transition select-none"
  >
    <div
      class="grow h-2/3"
    >
      <div
        class="text-center w-full h-full flex flex-col justify-center"
      >
        <span
          class="undefined font-text text-snowStorm-2 text-xs sm:text-sm md:text-base lg:text-lg leading-snug inline-block w-full max-h-full rounded px-2 py-1"
        />
      </div>
    </div>
    <div
      class="w-1/3 h-1/3"
    >
      <img
        alt=""
        src=""
      />
    </div>
  </div>
</DocumentFragment>
`;
